### PR TITLE
Parallel follow up to: Use eslint and @kentcdodds' presets

### PR DESCRIPTION
**Pull request by ta2edchimp**
Example solution to the original PR's [latest issue](https://github.com/gtramontina/ghooks/pull/43#issuecomment-202358212).

I made this PR directly towards `master` to trigger Travis.

### CODE REVIEW

**Comment by ta2edchimp** [comment]
>I made this PR directly towards master to trigger Travis.

... as well as to make it as effortless as possible to finish this guy, if the solution seems acceptable :wink:

**Comment by kentcdodds** [comment]
I think we should `.gitignore` dist.

**Comment by ta2edchimp** [comment]
Well, but then we'd have that egg and hen problem described by @gtramontina [here](https://github.com/gtramontina/ghooks/pull/43#issuecomment-202358212) again, don't we?
When Travis clones the repo and invokes `npm install`, it will fail, due to `./dist/*` missing. We'd be in the exact same position, #43 is atm.

Another idea I had this morning was, to utilize `opt-cli` once more. That is, providing an `.opt-out` rule that prevents running the package's `install` script on Travis.
But we'd have to put the `.opt-out` file into git and that would eventually interfere with any local settings from a contributor's working environment.

**Comment by kentcdodds** [comment]
Let me think on it... I'm sure we can do something else

**Comment by ta2edchimp** [comment]
I think I have an idea:

I extend `opt-cli` to also take environment variables into account. This way, we could specify an env var in Travis, say sth like

```
export OPT_OUT="ghooks-install"
```

and change the ghooks' package.json to:

```JSON
"scripts": {
    "install": "opt-cli --out ghooks-install --exec \"node ./bin/install\""
  }
```

This way, the script won't be invoked at all during a Travis build.

What do you think?

**Comment by kentcdodds** [comment]
That'd be cool.

Here's another idea. How about we take environment variables into account when running the install script?

**Comment by kentcdodds** [comment]
Sorry, I'm taking care of the kids right now. I think that won't work. Let's go for the opt-cli thing :-)

**Comment by kentcdodds** [comment]
Though we could probably just reference the existing CI environment variable rather than make our own.

**Comment by kentcdodds** [comment]
Actually, I kind of like the OPT_OUT environment variable. Values separated by commas or semicolons maybe?

**Comment by ta2edchimp** [comment]
>Actually, I kind of like the OPT_OUT environment variable. Values separated by commas or semicolons maybe?

Making these changes right now :laughing:

**Comment by ta2edchimp** [comment]
Can we somehow manually trigger a build on Travis?
I think I was too slow when adding `OPT_OUT` as an environment variable :( whenever I hit »Restart Build« Travis won't take the configured env var into account ...

**Comment by ta2edchimp** [comment]
Finally ... 🎉

```
Setting environment variables from repository settings
$ export OPT_OUT=ghooks-install
```
...
```
Done. Your build exited with 0.
```

File: package.json [hunk 1]
```diff
@@ -37,18 +39,22 @@
...
     "ghooks": "*",
-    "istanbul": "0.4.1",
-    "jshint": "^2.6.0",
-    "jshint-stylish": "^2.0.0",
+    "istanbul": "^1.0.0-alpha",
     "mocha": "^2.1.0",
     "mock-fs": "^3.0.0",
-    "opt-cli": "1.1.1",
+    "opt-cli": "1.4.0",
     "proxyquire": "^1.3.1",
```
Comment (kentcdodds) [inline package.json]: Should this be a dependency now because we use it on install?

**Comment by ta2edchimp** [comment]
`opt-cli` is now a general dependency.

**Comment by ta2edchimp** [comment]
>Actually, I kind of like the OPT_OUT environment variable. Values separated by commas or semicolons maybe?

To answer that question, `opt-cli` uses `path.delimiter` to split the env vars' values. This way *nix and Windows users may use the delimiter they're used to (for example to separate their `PATH` var's value) based on their system (colon, or semicolon respectively).

**Comment by gtramontina** [comment]
This is great! Thank you so much folks! I really appreciate it! :beers:

> ... as well as to make it as effortless as possible to finish this guy, if the solution seems acceptable :wink: 

Done!  :smile:

**Comment by gtramontina** [comment]
Hum… it seems now that, locally, I have to run `OPT_OUT=ghooks-install npm install` as well… ¬.¬:thought_balloon:

**Comment by ta2edchimp** [comment]
Aah, damn! Yeah, we `gitignored` the `dist` folder again m(
I was so excited about the env var idea that I totally forgot about `npm install`ing after fetching from git etc :disappointed: 
So, the `install` script should look for a `dist` folder and run `npm build` if it's not present.

**Pull request merged**